client/locations-list-map.js

```diff
@@ -1957,6 +1951,60 @@ const initGlobalAlert = () => {
   });
 };
 
+const initEmbedListEventListener = () => {
+  // Web toggle button
+  document.getElementById('js-ToggleListButton').addEventListener('click', toggleEmbedList);
+
+  // Mobile view toggle buttons
+  const mobileViewButtons = document.getElementsByClassName('emb-toggleButton');
+
+  for (let i = 0; i < mobileViewButtons.length; i++) {
+    mobileViewButtons[i].addEventListener('click', () => toggleMobileView(i));
+  }
+};
+
+const toggleEmbedList = () => {
+  const button = document.getElementById('js-ToggleListButton');
+  const container = document.getElementById('js-EmbedContainer');
+
+  if (container.classList.contains('list-open')) {
+    button.innerText = 'Open list';
+  } else {
+    button.innerText = 'Close list';
+  }
+
+  container.classList.toggle('list-open');
+};
+
+const toggleMobileView = (index) => {
+  const container = document.getElementById('js-EmbedContainer');
+  const webButton = document.getElementById('js-ToggleListButton');
+
+  const mobileViewButtons = document.getElementsByClassName('emb-toggleButton');
+
+  // If the data type is list we toggle open the list
+  if (mobileViewButtons[index].dataset.type === 'list') {
+    container.classList.add('list-open');
+  } else {
+    container.classList.remove('list-open');
+  }
+
+  // Set the web button text as well in case they have shrunk the screen
+  if (container.classList.contains('list-open')) {
+    webButton.innerText = 'Open list';
+  } else {
+    webButton.innerText = 'Close list';
+  }
+
+  for (let i = 0; i < mobileViewButtons.length; i++) {
+    if (index !== i) {
+      mobileViewButtons[i].dataset.active = 'false';
```

Comment:
any reason you're using strings here instead of booleans?

Reply:
I thought because of the data-active attribute ="true" it needs to equal a string or it throws an error, how else would you do this, I couldn't find anything on data-* could be a boolean value.

Reply:
ah I see, didn't quite get that this was added to the button as a data attribute, makes sense